### PR TITLE
Add possibility to configure radvd valid lifetime

### PR DESCRIPTION
**Pull request by schinken**

### CODE REVIEW

**Comment by pfSense-git-updates** [comment]
Before this pull request can be accepted you must first sign a CLA as described at https://www.pfsense.org/about-pfsense/#cla. Please read for more details.

File: src/etc/inc/services.inc [hunk 1]
```diff
@@ -181,6 +181,17 @@ function services_radvd_configure($blacklist = array()) {
 				$radvdconf .= "\t\tAdvRouterAddr on;\n";
 				break;
 		}
+
+		$validlifetime = $dhcpv6ifconf['ravalidlifetime'];
+		if(is_numeric($validlifetime)) {
+		  $radvdconf .= "\t\tAdvValidLifetime {$validlifetime};\n";
```
Comment (rbgarga) [inline src/etc/inc/services.inc]: IMO you don't need to set $validlifetime here, you can check using array value direct and using is_numericint() to make sure it's a positive number
```
if (is_numericint($dhcpv6ifconf['ravalidlifetime'])) {
```

File: src/etc/inc/services.inc [hunk 2]
```diff
@@ -181,6 +181,17 @@ function services_radvd_configure($blacklist = array()) {
...
+		}
+
+		$preferredlifetime = $dhcpv6ifconf['rapreferredlifetime'];
+		if(is_numeric($preferredlifetime)) {
+		  $radvdconf .= "\t\tAdvPreferredLifetime {$preferredlifetime};\n";
```
Comment (rbgarga) [inline src/etc/inc/services.inc]: Same suggestions here

**Comment by schinken** [comment]
Changed the pull request according to your feedback

**Comment by davidjwood** [comment]
Point taken on the defaults, but the possibility exists that the default in radvd will change silently or does not conform with the documentation, at which point the help text is misleading. If the default is configured explicitly in an else {} block, there is no ambiguity as the default you claim will be 'baked in' to the generated radvd configuration.

**Comment by schinken** [comment]
Yeah okay, you're right. I'll do another pull request soon

**Pull request merged**